frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/highlighting/MacheteAnnotator.java

```diff
@@ -80,6 +87,34 @@ private void processMacheteGeneratedEntry(MacheteGeneratedEntry macheteEntry, An
                   .format(processedBranchName))
           .range(branch).create();
     }
+
+    // update the state of the .git/machete VirtualFile so that new entry is available in the VirtualFile
+    ModalityUiUtil.invokeLaterIfNeeded(NON_MODAL, () -> saveDocumentBeforeCheck(file));
+    // check for duplicate entries in the machete file
+    try {
+      val branchNamesFromFile = VfsUtil.loadText(file.getVirtualFile());
+      if (isBranchNameRepeated(branchNamesFromFile, processedBranchName)) {
+        holder.newAnnotation(HighlightSeverity.ERROR,
+            getNonHtmlString("string.GitMachete.MacheteAnnotator.branch-entry-already-defined")
+                .format(processedBranchName))
+            .range(branch).create();
+      }
+    } catch (PluginException | IllegalStateException ignored) { // ignore dubious IDE checks against annotation range
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @UIEffect
+  private void saveDocumentBeforeCheck(PsiFile file) {
+    val fileDocManager = FileDocumentManager.getInstance();
+    fileDocManager.saveDocument(Objects.requireNonNull(fileDocManager.getDocument(file.getVirtualFile())));
```

Comment:
Ok one more thing, from javadoc of `com.intellij.openapi.fileEditor.FileDocumentManager#getDocument`:

```
Should be invoked in a read action.
```

Either fix it now (https://plugins.jetbrains.com/docs/intellij/general-threading-rules.html#read-write-lock) or at least add a `TODO` for https://github.com/VirtusLab/git-machete-intellij-plugin/issues/1227

Reply:
Hm, for me it looks like current logic is valid in that aspect. Reasoning:
1. From the https://plugins.jetbrains.com/docs/intellij/general-threading-rules.html#read-write-lock guide:
> Reading data is allowed from any thread. Reading data **from the UI thread does not require any special effort.**
2. `saveDocumentBeforeCheck` method is annotated with `@UIEffect` and (in effect) it runs on the Event Dispatch Thread (checked it with SwingUtilities#isEventDispatchThread)
3. So, it looks like  `com.intellij.openapi.fileEditor.FileDocumentManager#getDocument` called inside `saveDocumentBeforeCheck` reads data from the UI thread, and thus it does not have to be wrapped in a special read action.

Does it make sense?

Reply:
Awww okay good spot, I forgot that on UI thread one doesn't need ReadActions. So good to go as is